GestureHandler.web.js

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import {
+  ScrollView,
+  Slider,
+  Switch,
+  TextInput,
+  ToolbarAndroid,
+  ViewPagerAndroid,
+  DrawerLayoutAndroid,
+  FlatList,
+  TouchableWithoutFeedback,
+} from 'react-native';
+import gestureHandlerRootHOC from './gestureHandlerRootHOC';
+import Directions from './Directions';
+import State from './State';
+
+// Factory for stub Handler components
+function createStubHandler(name) {
+  return class NativeViewGestureHandler extends React.Component {
+    static displayName = name;
+
+    setNativeProps() {
+      // Since this is a stub we do not need to pass on native props.
+      // However, we need to implement it here to avoid null calls.
+    }
+
+    render() {
+      const { children, ...rest } = this.props;
+
+      // We don't want to create another layer, so instead we just clone it
+      const child = React.Children.only(children);
+      return React.cloneElement(child, {
+        ...rest,
+      });
+    }
+  };
+}
+
+// Create all Handler components with their respective handler functions
+// (at the moment only TapGestureHandler is properly supported)
+const NativeViewGestureHandler = createStubHandler('NativeViewGestureHandler');
+
+class TapGestureHandler extends React.Component {
+  setNativeProps() {}
+
+  handlePress = ({ nativeEvent: { locationX, locationY, pageX, pageY } }) => {
+    const { enabled, onHandlerStateChange } = this.props;
+
+    if (enabled !== false && onHandlerStateChange) {
+      onHandlerStateChange({
+        nativeEvent: {
+          oldState: State.ACTIVE,
+          state: State.UNDETERMINED,
+          x: locationX,
+          y: locationY,
+          absoluteX: pageX,
+          absoluteY: pageY,
+        },
+      });
+    }
+  };
+
+  render() {
+    const { children, maxDurationMs, style } = this.props;
+
+    return (
+      <TouchableWithoutFeedback
```

Comment:
And can't we use `onPressIn` and `onPressIn` for achieving more native-like states' flow?